src/builders/indexable-builder.php

```diff
@@ -403,6 +414,9 @@ public function build( $indexable, $defaults = null ) {
 			return $this->save_indexable( $indexable, $indexable_before );
 		}
 		catch ( Source_Exception $exception ) {
+			if ( ! $this->is_type_with_no_id( $indexable->object_type ) && ( ! isset( $indexable->object_id ) || \is_null( $indexable->object_id ) ) ) {
```

Comment:
To be on the safe side, we could consider moving this condition to the `save_indexable` method. That way, the behaviour doesn't change in other code paths (as we don't have to return `false` here anymore)

Reply:
On second thought, having indexables with a `NULL` id is worse than not having an indexable at all. Please discard my comment

Reply:
Yeah, and we already return `false` when throwing a `Not_Built_Exception` later on, so it's basically the same case :)